### PR TITLE
Fix nix edit

### PR DESCRIPTION
**Pull request by Mic92**
# Motivation

Since we introduced SourceAccessor, nix edit no longer works as described in https://github.com/NixOS/nix/issues/9652

This pull requests fixes the issue and adds regression test so that we no longer have this issue in future.

# Context

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: tests/functional/simple.nix [hunk 1]
```diff
@@ -5,4 +5,5 @@ mkDerivation {
   builder = ./simple.builder.sh;
   PATH = "";
   goodPath = path;
+  meta.position = "${__curPos.file}:${toString __curPos.line}";
```
Comment (Mic92) [inline tests/functional/simple.nix:8]: I also saw that we didn't test this anywhere.

File: tests/functional/flakes/flake.nix
Comment (roberth) [inline tests/functional/flakes/flake.nix:1]: Is this file needed? It looks like it was added by accident.

**Review by roberth — COMMENTED** [review]
LGTM, but I don't think `tests/functional/flakes/flake.nix` is needed.

**Review by edolstra — APPROVED** [review]

File: tests/functional/flakes/flake.nix
Comment (Mic92) [reply tests/functional/flakes/flake.nix:1]: Right.

**Review by roberth — APPROVED** [review]

**Comment by kanashimia** [comment]
Thanks, this was a blazingly fast fix :heart:
